### PR TITLE
feat: store user email

### PR DESCRIPTION
**Pull request by Harish-osmosys**
### Pre-requisites

- [x] I have gone through the Contributing guidelines for [Submitting a Pull Request (PR)](../../CONTRIBUTING.md#submitting-a-pull-request-pr) and ensured that this is not a duplicate PR.
- [x] I have performed preliminary testing to ensure that any existing features are not impacted and any new features are working as expected.

### PR Details

PR details have been updated as per the given format (see below)

- [x] PR title adheres to the format specified in guidelines (e.g., `feat: add admin login page`)
- [x] Description has been added
- [x] Related changes have been added (optional)
- [x] Screenshots have been added (optional)
- [ ] Pending actions have been added (optional)
- [ ] Any other additional notes have been added (optional)

### Additional Information

- [x] Appropriate label(s) have been added (`ready for review` should be added if the PR is ready to be reviewed)
- [x] Assignee(s) and reviewer(s) have been added (optional)

---

**Description:**

- This Pr allow to store the user email in the createdBy field to track the activities
- Enhance the CreateSummary resolver to update the createdBy field with the current user's email ID obtained from Azure AD authentication.

**Related changes:**

- Add custom decorator : CurrentUser 
- Update the Azure-ad Strategy 

**Screenshots:**

image

**Pending actions:**

**Additional notes:**

## Summary by CodeRabbit

- **New Features**
  - Implemented user context retrieval with the `CurrentUser` decorator for enhanced user-specific operations.
  - Enhanced summary creation to include user email, setting `createdBy` and `modifiedBy` fields for better tracking.

- **Improvements**
  - Updated authentication strategy to store additional user information, improving user data handling during login.

These updates aim to personalize user interactions and improve data management within the application.

### CODE REVIEW

File: apps/api/src/auth/azure-ad.strategy.ts [hunk 1]
```diff
@@ -23,6 +23,12 @@ export class AzureADStrategy extends PassportStrategy(BearerStrategy, 'azure-ad'
   async validate(profile: any): Promise<any> {
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply apps/api/src/auth/azure-ad.strategy.ts:23]: I am not sure about this , but i think we are not sure what field we will be using so kept this dynamic by using any for now

File: apps/api/src/modules/summary/summary.service.ts [hunk 2]
```diff
@@ -55,6 +55,11 @@ export class SummaryService extends CoreService<Summary> {
               this.logger.log(
                 `Saving Uploaded File Details for ${modifiedFilename} at ${fileLocation}`,
               );
+
+              if (userEmail) {
+                summary.createdBy = userEmail;
```
Comment (xixas) [inline apps/api/src/modules/summary/summary.service.ts:60]: Update Modifed By too here. Initial one will be both user, later changes done in backend can be kept as done by system later.

File: apps/api/src/auth/azure-ad.strategy.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (xixas) [reply apps/api/src/auth/azure-ad.strategy.ts:23]: No need

**Review by xixas — CHANGES_REQUESTED** [review]

File: apps/api/src/modules/summary/summary.service.ts (hunk 2 above)
Comment (Harish-osmosys) [reply apps/api/src/modules/summary/summary.service.ts:60]: updated in https://github.com/OsmosysSoftware/osm-transcript-summarizer/pull/37/commits/ae042a610ff3b1e92f5c0584331d2314df0a57fd

**Review by xixas — APPROVED** [review]

**Pull request merged**